### PR TITLE
add github-workflow building wheels

### LINKED ISSUE #123: Wheels on Pypi
Please consider uploading the wheels to PyPI!
It would be realy helpful.

### PR DESCRIPTION
**Pull request by betaboon**
resolves #123 

Hello.

This PR adds a github-workflow to create wheels containing all the required libraries.
builds include:
- python: 3.8-3.11
- os: linux and macos

This now builds `leptonica` and `tesseract` from source.

~~some further information tho:
since `tesseract` and `leptonica` are being installed via `yum` (for `manylinux`), `apk` (for `musllinux`) and `brew` (for `macos`) all of the built wheels contain  a different version of tesseract.
that might pose a problem to some.
i think the only way to get around that would be to compile tesseract from source.~~

### CODE REVIEW

**Comment by betaboon** [comment]
@zdenop i think i addressed all your review comments now.

**Comment by zdenop** [comment]
I like your build scripts, but I am not sure if there should be part of tesserocr. Especially if not experienced user would use them.
Here are some comments from perspective of tesserocr builder/user:
1. On linux and Mac you should first check if tesseract (any probably leptonica) is not already installed. if yes then it should be uninstalled first before building tesseract/leptonica to avoid further problems.
2. If your script should be used for creating tesserocr python package/wheel - you should stick to system provided leptonica/tesseract (API/ABI of different version are not the same and programs linked against shared libraries could be broken if different version is present). Other solution could be that you include your custom build shared libraries to tesseocr wheel as [it is done for windows](https://github.com/simonflueckiger/tesserocr-windows_build/releases)
3. `pango`,  `cairo` and  `icu4c` (installed for MacOS) are needed only for training tools - tesserocr is not able to use them e.g.  scripts (also for linux) should build tessaract without training tools if you build it only for tesserocr
4. Personally I prefer to build leptonica with zlib and png support (AKA minimalistic build) - tesserocr is used for OCR, and python is able to open rest of image formats (with PIL or OpenCV). This decrease tesserocr dependency complexity...

**Comment by zdenop** [comment]
Regarding github action part - this something I love to see in tesserocr (with windows weels - but that is more difficult part).

**Comment by zdenop** [comment]
For building release wheels use tesseract and leptonica provided by system.

If you plan to introduce testing CI (e.g. for commits, PR) then integrate those scripts to github actions as [tesseract](https://github.com/tesseract-ocr/tesseract/blob/main/.github/workflows/cmake.yml) and [leptonica](https://github.com/DanBloomberg/leptonica/blob/master/.github/workflows/cmake.yml) do.

> do i get this right: you're suggesting to compile tesseract with -DBUILD_TRAINING_TOOLS=OFF and removing those dependencies?

yes

> do i get this right: you're suggesting to to compile leptonic with -DENABLE_GIF=OFF, -DENABLE_JPEG=OFF, -DENABLE_TIFF=OFF, -DENABLE_WEBP=OFF and -DENABLE_OPENJPEG=OFF and removing those dependencies?

yes.

**Comment by betaboon** [comment]
so i just pushed changes doing the following:
- move the buildscripts to `.github` as to signify that they are only being used for ci
- build tesseract with `-DBUILD_TRAINING_TOOLS=OFF` and removing the appropriate dependencies
- build leptonic with the discussed flags and removing the dependencies

is there anything left for me to do here to get this merged?
eg a ci-job to publish to pypi?

**Review by sirfz — APPROVED** [review]
Looks great to me, thank you for taking the time to do this. Awaiting @zdenop's approval before merging

**Review by zdenop — APPROVED** [review]

**Comment by sirfz** [comment]
hi @betaboon, I'll include the wheels with the next release as I can't modify an existing release on pypi

**Comment by betaboon** [comment]
do you have any idea when that might be ?

**Pull request merged**
